src/modals/examples/modal-overlay.js

```diff
@@ -0,0 +1,203 @@
+/**
+ * @ngdoc directive
+ * @name patternfly.modals.componenet:pfModalOverlay
+ * @element pfModalOverlay
+ * @restrict E
+ *
+ * @description
+ * The Modal Overlay pattern provides a way to quickly load and display important information to the user without navigating
+ * away from the current page. It utilizes the {@link https://angular-ui.github.io/bootstrap/#modal Angular UI modal}
+ * and Patternfly styling. Use cases for Modal Overlays can vary a great deal, but some examples include the following:
+ * <ul>
+ * <li>To remind or prompt users
+ * <li>To load dialogs that require user input before advancing
+ * <li>To load tasks which require a user’s full attention, such as stepping through a wizard flow
+ * <li>To present important information or warnings
+ * </ul>
+ *
+ * @param {boolean} showModal Flag for setting modal to hide/show by default
+ * @param {function} onClose Function to call when the modal is closed<br/>
+ * Note: since 'on-close' is a '&', param passed must be named 'dismissCause'
+ * @param {string} modalId Id of the modal
+ * @param {string} modalTitle The title of the modal displayed in the header
+ * @param {string} modalBodyTemplate Path to html template for modal body
+ * @param {array} actionButtons array of button objects. Each button can have the following properties:<br/>
+ * <ul style='list-style-type: none'>
+ * <li>.label        - the text to display on the button
+ * <li>.class        - (optional) classes to add to the button
+ * <li>.actionFn     - (optional) user defined function to call when the button is clicked. Should return true to close modal, otherwise should return false.
```

Comment:
Pls remove: '...Should return true to close modal, otherwise should return false.'.   I don't think we need to support validation after Ok button is clicked, if modal is invalid, Ok button should be disabled.